core/templates/dev/head/pages/exploration_player/PlaythroughService.js

```diff
@@ -0,0 +1,275 @@
+          playthrough.playthroughId = response.playthroughId;
+        });
+      }
+    };
+
+    var getFullPlaythroughUrl = function() {
+      return UrlInterpolationService.interpolateUrl(
+        STORE_PLAYTHROUGH_URL, {
+          exploration_id: playthrough.expId
+        });
+    };
+
+    return {
+      initSession: function(newExplorationId, newExplorationVersion) {
+        playthrough = PlaythroughObjectFactory.createNew(
+          null, newExplorationId, newExplorationVersion, null, {}, []);
+        expStopwatch = StopwatchObjectFactory.create();
+      },
+      getPlaythrough: function() {
+        return playthrough;
+      },
+      recordExplorationStartAction: function(initStateName) {
+        playthrough.actions.push(LearnerActionObjectFactory.createNew(
+          ACTION_TYPE_EXPLORATION_START,
+          {
+            state_name: {
+              value: initStateName
+            }
+          },
+          CURRENT_ACTION_SCHEMA_VERSION
+        ));
+
+        multipleIncorrectStateName = {
+          state_name: initStateName,
+          num_times_incorrect: 0
+        };
+
+        cycleIdentifier = {
+          cycle: '',
+          num_cycles: 0
+        };
+        visitedStates.push(initStateName);
+      },
+      recordAnswerSubmitAction: function(
+          stateName, destStateName, interactionId, answer, feedback,
+          timeSpentInStateSecs) {
+        playthrough.actions.push(LearnerActionObjectFactory.createNew(
+          ACTION_TYPE_ANSWER_SUBMIT,
+          {
+            state_name: {
+              value: stateName
+            },
+            dest_state_name: {
+              value: destStateName
+            },
+            interaction_id: {
+              value: interactionId
+            },
+            answer: {
+              value: answer
+            },
+            feedback: {
+              value: feedback
+            },
+            time_spent_in_state_secs: {
+              value: timeSpentInStateSecs
+            }
+          },
+          CURRENT_ACTION_SCHEMA_VERSION
+        ));
+
+        if (destStateName === stateName) {
+          incrementIncorrectAnswerInMultipleIncorrectIssueTracker();
+        } else {
+          handleStateTransitionInMultipleIncorrectIssueTracker(destStateName);
+
+          handleStateTransitionInCyclicIssueTracker(destStateName);
+        }
+      },
+      recordExplorationQuitAction: function(
+          stateName, timeSpentInStateSecs) {
+        playthrough.actions.push(LearnerActionObjectFactory.createNew(
+          ACTION_TYPE_EXPLORATION_QUIT,
+          {
+            state_name: {
+              value: stateName
+            },
+            time_spent_in_state_secs: {
+              value: timeSpentInStateSecs
+            }
+          },
+          CURRENT_ACTION_SCHEMA_VERSION
+        ));
+      },
+      recordPlaythrough: function() {
+        // If playthrough ID exists, issue has already been identified and the
```

Comment:
nit: move these comments into the branches they're describing:

```javascript
if (...) {
  // playthroughId exists so issue has already...
} else {
  // check whether an issue...
}
```

Reply:
Done.